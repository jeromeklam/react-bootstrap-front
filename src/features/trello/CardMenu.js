import React from 'react';
import PropTypes from 'prop-types';
import { DropdownWrapper } from '../advanced';
import { DropdownMenu, DropdownMenuOption, DropdownMenuDivider } from '../basic';

const CardMenu = props => (
  <DropdownWrapper
    position="bottom"
    trigger={<button className="btn btn-sm btn-light trello-card-menu-menu">&hellip;</button>}
    tooltip="Actions Ã  effectuer sur la carte"
  >
    <DropdownMenu className="trello-card-menu">
      {props.update && (
        <DropdownMenuOption
          onClick={props.onUpdate}
          label={props.t({ id: 'rbf.trello.card.update', defaultMessage: 'Update card' })}
        />
      )}
      {props.options &&
        props.options.length > 0 &&
        props.options.map(option => (
          <DropdownMenuOption
            key={option.label}
            onClick={option.onClick}
            label={option.label}
            theme={option.theme || null}
          />
        ))}
      {(props.delete || props.remove) && <DropdownMenuDivider />}
      {props.remove && (
        <DropdownMenuOption
          onClick={props.onRemove}
          label={props.t({ id: 'rbf.trello.card.remove', defaultMessage: 'Remove card' })}
          theme="warning"
        />
      )}
      {props.delete && (
        <DropdownMenuOption
          onClick={props.onDelete}
          label={props.t({ id: 'rbf.trello.card.delete', defaultMessage: 'Delete card' })}
          theme="warning"
        />
      )}
    </DropdownMenu>
  </DropdownWrapper>
);

CardMenu.propTypes = {
  remove: PropTypes.bool,
  delete: PropTypes.bool,
  onRemove: PropTypes.func,
  onDelete: PropTypes.func,
  onUpdate: PropTypes.func,
  options: PropTypes.array,
  t: PropTypes.func.isRequired,
  update: PropTypes.bool,
};

CardMenu.defaultProps = {
  remove: true,
  delete: true,
  onRemove: () => {},
  onDelete: () => {},
  onUpdate: () => {},
  options: [],
  update: true,
};

export default CardMenu;
