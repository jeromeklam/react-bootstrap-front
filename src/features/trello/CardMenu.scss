.trello-card-menu {
  .trello-lane-card-menu {
    transition: all 0.5s ease;
    display: inline-block;
    border: none;
    outline: none;
    font-size: 16px;
    font-weight: bold;
    height: 15px;
    line-height: 1px;
    margin: 0 0 8px;
    padding: 0;
    text-align: center;
    width: 15px;
    background: inherit;
    cursor: pointer;
  }
}
