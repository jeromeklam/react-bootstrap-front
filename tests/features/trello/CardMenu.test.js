import React from 'react';
import renderer from 'react-test-renderer';
import CardMenu from '../../../src/features/trello/CardMenu';

const callback = jest.fn();

it('renders standard', () => {
  // First render
  const component = renderer.create(
    <CardMenu t={callback} />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});
