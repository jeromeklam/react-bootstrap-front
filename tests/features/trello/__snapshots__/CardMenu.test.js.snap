// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders standard 1`] = `
<div
  className="advanced-dropdown-wrapper"
  onClick={[Function]}
>
  <button
    className="btn btn-sm btn-light trello-card-menu-menu"
  >
    â€¦
  </button>
</div>
`;
